### PR TITLE
Fix #307

### LINKED ISSUE #307: High Priority - Verify Block
We have the  transaction/block processor built, but it's not doing any kind of validation.
The block should be verified before being processed.

### PR DESCRIPTION
**Pull request by melanke**

### CODE REVIEW

File: src/NeoSharp.Core/Messaging/Handlers/BlockMessageHandler.cs [hunk 1]
```diff
@@ -50,10 +53,15 @@ public override bool CanHandle(Message message)
...
             var block = message.Payload;
-
+            
             if (block.Hash == null)
             {
-                this._blockOperationsManager.Sign(block);
+                this._blockSigner.Sign(block);
+            }
+
+            if (_blockVerifier.Verify(block))
```
Comment (osmirnov) [inline src/NeoSharp.Core/Messaging/Handlers/BlockMessageHandler.cs:62]: I guess we can save a bit here - no need in verification if we just signed a block.

**Review by osmirnov — APPROVED** [review]

File: src/NeoSharp.Core/Messaging/Handlers/BlockMessageHandler.cs (hunk 1 above)
Comment (shargon) [reply src/NeoSharp.Core/Messaging/Handlers/BlockMessageHandler.cs:62]: Will be verified before?

Comment (osmirnov) [reply src/NeoSharp.Core/Messaging/Handlers/BlockMessageHandler.cs:62]: I meant if the hash was null on line 58 and we signed on line 59. It makes a little sense to verify on line 62.

**Comment by aboimpinto** [comment]
In my PR I put everything back to the OperationManager
we should not have two classes for the same concern. The Signature and the Verification should be in the same place because most of the cases they share the same dependencies.

The solution for the Circular References is not splitting this functionalities but remove the IBlockchain and IServer from the MessageHandlers and I did that on my pending PR.

I've been working on this for the last 2 days as I said in Discord

image

I already revert the changes for the transactions on my pending PR

**Pull request merged**